### PR TITLE
Cancelable edit

### PR DESCRIPTION
**Pull request by nelix**
screen shot 2015-02-05 at 3 37 46 pm
It restores the original value on pressing escape or clicking cancel.
It raises an event (onSave) when pressing meta+enter or clicking Save {this.props.displayName}
It warns you when cancelling
It optionally allows blank input
Edit mode is triggered on click
It optionally grows with input

### CODE REVIEW

File: packages/cancelable-edit/index.jsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  },
+
+  handleChange(e) {
+    this.setState({
+      pendingValue: e.target.value,
+    });
+  },
+
+  handleCancel(e) {
+    if (this.props.value === this.state.pendingValue ||
```
Comment (nelix) [inline packages/cancelable-edit/index.jsx]: this.unsaved() could be used here

File: packages/cancelable-edit/index.jsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+      confirm(this.props.warnMessage)
```
Comment (nelix) [inline packages/cancelable-edit/index.jsx]: Would it be better to use our own pretty dialog here?

Comment (ojame) [inline packages/cancelable-edit/index.jsx]: Was just about to ask the same thing, probably? Wish our api for our alert was much, much cleaner :(.

**Comment by grapegravity** [comment]
The Cancel button isn’t working for me in the example

**Comment by nelix** [comment]
should be fixed now

File: packages/cancelable-edit/style.scss [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  background-color: $color-cancelable-edit-background;
+  outline: 0;
+  resize: none;
+  width: 100%;
+  margin-top: 10px;
+  border-radius: 3px;
+  padding: 10px;
+  cursor: text;
+
+  &[readonly="readonly"], &[readonly] {
```
Comment (grapegravity) [inline packages/cancelable-edit/style.scss]: Correct me if I’m mistaken, but `&[readonly]` should itself match any `readonly` values given?

Comment (ojame) [inline packages/cancelable-edit/style.scss]: `&[readonly]` will target any true bool, like `<input readonly />`. `&[readonly="readonly"]` will target `<input readonly="readonly" />`

Comment (grapegravity) [inline packages/cancelable-edit/style.scss]: Just checked and `&[readonly]` will target `readonly` being present, with or without a value.

screen shot 2015-02-06 at 10 32 07 am

Comment (ojame) [inline packages/cancelable-edit/style.scss]: i'm pretty sure this is a firefox thing, let me check

Comment (grapegravity) [inline packages/cancelable-edit/style.scss]: http://codepen.io/grapegravity/pen/YPEWvo seems to work the same in Firefox

Comment (ojame) [inline packages/cancelable-edit/style.scss]: looks like it's a ye old problem that no longer exists: http://stackoverflow.com/questions/9635293/how-to-style-readonly-attribute-with-css

**Pull request merged**